fire/io/bernese/bernese.py

```diff
@@ -0,0 +1,314 @@
+            ["STATION 1", "XYZ 1", "STATION 2", "XYZ 2", "MATRIX ELEMENT"], params
+        )
+    )
+
+
+class BerneseSolution(object):
+    resultatset: BerneseSet
+
+    def __init__(self, addneq_fil: Path, crd_fil: Path, cov_fil: Path) -> None:
+        """
+        Læs og ekstraher Bernese data fra et sæt af matchende ADDNEQ, CRD og (valgfrit) COV filer
+
+        Pointen er at skabe en liste med Station'er baseret på indholdet af filerne og tilknytte koordinater (CRD-fil),
+        kovarians matrix (COV-fil) og andre informationer (ADDNEQ-filen)
+
+        addneq_fil: sti til en Bernese ADDNEQ fil (påkrævet)
+        crd_fil   : sti til en Bernese CRD (påkrævet)
+        cov_fil   : sti til en Bernese COV (valgfri)
+
+        Uddatatype: fire.io.bernese.BerneseSet
+        """
+        self.resultatset: BerneseSet = BerneseSet()
+        self.resultatset.stations = {}
+
+        if not crd_fil.exists():
+            raise ValueError("CRD-fil skal angives!")
+        # Begynd med at indlæse opbygge stationer fra CRD-fil
+        with open(crd_fil, "r") as crd:
+            self.crd_parse(crd.readlines())
+
+        # Dernæst tilføj kovariansmatricer fra COV-fil hvis angivet
+        if cov_fil.exists():
+            with open(cov_fil, "r") as cov:
+                self.cov_parse(cov.readlines())
+
+        # Endelig tilføjes observationslængde og spredning fra ADDNEQ
+        if not addneq_fil.exists():
+            raise ValueError("ADDNEQ-fil skal angives!")
+        with open(addneq_fil, "r") as addneq:
+            self.addneq_parse(addneq.readlines())
+
+    def crd_parse(self, crd_data: list) -> None:
+        """
+        Parse header og datalinjerne fra en koordinat (CRD) fil og opret stationer
+        """
+        for linje_nr, line in enumerate(crd_data, start=1):
+            if linje_nr == 1:  # header med ugenr/comb week
+                self.resultatset.gnss_week = crd_parse_weekline(line)
+            elif linje_nr == 3:  # header med datum og epoke
+                headerlinje = crd_parse_epochdatumline(line)
+                self.resultatset.datum = headerlinje["DATUM"]
+                self.resultatset.epoch = strptime(
+                    headerlinje["EPOCH"], "%Y-%m-%d %H:%M:%S"
+                )
+            elif linje_nr >= 7:  # data begynder på linje 7 i CRD-filer
+                if not line.isspace():  # skip evt. tomme linjer
+                    datalinje = crd_parse_dataline(line)
+                    station = Station(
+                        navn=datalinje["STATION NAME"],
+                        flag=datalinje["FLAG"],
+                        koordinat=Koordinat(
+                            x=datalinje["X"], y=datalinje["Y"], z=datalinje["Z"]
+                        ),
+                        kovarians=None,
+                        spredning=None,
+                    )  # kovarians of spredning bliver sat senere
+                    self.resultatset.stations[datalinje["STATION NAME"]] = station
```

Comment:
Den her situation går igen en del steder. Brug `continue` hvor muligt fremfor at slæbe rundt på en ekstra niveau indentering. @busstoptaktik har en fin en anekdote om Taoistisk Programmering som han sikkert gerne underholder med engang ved lejlighed. Det korte af det lange er, ryd problemerne af vejen, og tag så hånd om den primære funktionalitet.

```suggestion
        for linje_nr, line in enumerate(crd_data, start=1):
            if linje_nr == 1:  # header med ugenr/comb week
                self.resultatset.gnss_week = crd_parse_weekline(line)
                continue
            if linje_nr == 3:  # header med datum og epoke
                headerlinje = crd_parse_epochdatumline(line)
                self.resultatset.datum = headerlinje["DATUM"]
                self.resultatset.epoch = strptime(
                    headerlinje["EPOCH"], "%Y-%m-%d %H:%M:%S"
                )
                continue
            if linje_nr < 7 or line.isspace():
                continue
                
            # Nu kan vi begynde at læse det egentlige indhold af filen
            datalinje = crd_parse_dataline(line)
            station = Station(
                navn=datalinje["STATION NAME"],
                flag=datalinje["FLAG"],
                koordinat=Koordinat(
                    x=datalinje["X"], y=datalinje["Y"], z=datalinje["Z"]
                ),
                kovarians=None,
                spredning=None,
            )  # kovarians of spredning bliver sat senere
            self.resultatset.stations[datalinje["STATION NAME"]] = station
```